src/imap/select_folder.rs

```diff
@@ -61,11 +61,13 @@ impl Imap {
 
     /// select a folder, possibly update uid_validity and, if needed,
     /// expunge the folder to remove delete-marked messages.
+    /// Returns true if a new folder was selected. If this returns false, then no SELECT command was run
+    /// and self.config.selected_mailbox was not updated (so, e.g. it may contain an outdated uid_next)
```

Comment:
this should probably be an enum, it is really hard to keep track of what these boolean return values mean

Reply:
Hmmm... What should the possible values be called then, such that it's more understandable for the reader than a bool? `NewlySelected` seems to be the obvious choice for the positive result but what's the other one? `SelectedMailboxContainsOutdatedData` is too long. Maybe `NotNewlySelected`? Or just call the enum `NewlySelected` and make options `Yes` and `No`?

I find the last option the most understandable one but I'll wait for you to comment as I also found a bool understandable :)

Reply:
NewlySelected::Yes/No sounds great

Reply:
Ok